mms/tests/integration_tests/test_utils.py

```diff
@@ -0,0 +1,193 @@
+        url,
+        onnx_model,
+        onnx_source_model_zoo=True,
+        is_onnx_model=True):
+    # Download the files required for onnx integ tests.
+    download_dir = tmpdir + '/mxnet_model_server'
+    try:
+        if is_onnx_model:
+            os.mkdir(download_dir)
+            if onnx_source_model_zoo:
+                _download_file(download_dir, url[onnx_model])
+                _download_file(
+                    download_dir,
+                    "https://s3.amazonaws.com/model-server/models/onnx-squeezenet/signature.json")
+                _download_file(
+                    download_dir,
+                    "https://s3.amazonaws.com/model-server/models/onnx-squeezenet/synset.txt")
+                model_tar = '{}/{}.tar.gz'.format(download_dir, onnx_model)
+                tar = tarfile.open(model_tar, "r:*")
+                tar.extractall(path=download_dir)
+                tar.close()
+                model_dir = '{}/{}'.format(download_dir, onnx_model)
+                model_path = os.path.join(model_dir, 'model.onnx')
+                new_path = os.path.join(
+                    model_dir, '{}.onnx'.format(onnx_model))
+                os.rename(model_path, new_path)
+                shutil.move(new_path, download_dir)
+
+            else:
+                _download_file(download_dir, url[onnx_model])
+                _download_file(
+                    download_dir,
+                    "https://s3.amazonaws.com/model-server/models/" +
+                    onnx_model +
+                    "/signature.json")
+                _download_file(
+                    download_dir,
+                    "https://s3.amazonaws.com/model-server/models/" +
+                    onnx_model +
+                    "/synset.txt")
+            print('model files prepared for model {} '.format(onnx_model))
+            # Export the model.
+            print("Exporting the mxnet model server model...")
+            subprocess.check_call(['mxnet-model-export',
+                                   '--model-name',
+                                   onnx_model,
+                                   '--model-path',
+                                   download_dir],
+                                  cwd=download_dir)
+            shutil.move('{}/{}.model'.format(download_dir, onnx_model), tmpdir)
+            for root, dirs, files in os.walk(download_dir):
+                for f in files:
+                    os.unlink(os.path.join(root, f))
+                for d in dirs:
+                    shutil.rmtree(os.path.join(root, d))
+            shutil.rmtree(download_dir)
+        else:
+            _download_file(tmpdir, url[onnx_model])
+    except Exception as e:
+        print("Failed to create models. {}".format(str(e)))
+        raise
+
+
+def start_test(
+        tmpdir,
+        url,
+        onnx_model,
+        port='8080',
+        onnx_source_model_zoo=False,
+        is_onnx_model=True,
+        test_multiple_models=False):
```

Comment:
instead of passing in a dict of all models, and an (optional) model name, and a flag for multiple models, just change the function so you take a list (or dict if necessary) containing the models you are *actually* testing.